lib/src/components/Tab/Tab.tsx

```diff
@@ -0,0 +1,86 @@
+import { assignInlineVars } from "@vanilla-extract/dynamic";
+import { useEffect, useState } from "react";
+import { Badge } from "../Badge";
+import {
+  badgeStyles,
+  contentStyles,
+  iconStyles,
+  tabHeaderContainerStyles,
+  tabRecipe,
+  tabVars,
+} from "./tab.css";
+import { TabsObjectProps, TabProps } from "./tab.types";
+
+export const Tab = ({
+  defaultValue,
+  value,
+  color = "#2F80ED",
+  background = "#d7e9ff",
+  onChange = () => {},
+  grow = false,
+  height = "40px",
+  tabs,
+  style,
+  className,
+  children,
+  ...nativeProps
+}: TabProps) => {
+  const [selectedTab, setSelectedTab] = useState<TabsObjectProps>(
+    {} as TabsObjectProps
+  );
+
+  useEffect(() => {
+    const comparator = value || defaultValue;
```

Comment:
`comparator` also does not tell the purpose, could we name this as `selectedTabValue`